### PR TITLE
add CI via GitHub actions

### PR DESCRIPTION
**Pull request by Bogdanp**
I noticed how much additional work my last PR required (sorry @jeapostrophe !) to get working on drdr so I figured this repo could use CI of its own. [GH Actions](https://github.com/features/actions) are turning GA on November 13 and they're pretty convenient for this sort of thing.

You can see an example CI run [here](https://github.com/bogdanp/web-server/commit/c7855c4b3652190ff84a7803e11e9b01fbe27cb3/checks?check_suite_id=292872259).

EDIT: I added another commit to make the e2e tests wait for each server to become available before running their tests.

### CODE REVIEW

**Comment by jeapostrophe** [comment]
Excellent. Thanks, I hadn't yet had the time to look at what the new failure was.

And no worries about the changes I had to do, it was nothing compared to what you did!

File: README.md [hunk 1]
```diff
@@ -1,10 +1,17 @@
 # Racket web-server
 
-This is the main repository for the Racket web-server package.  
-  
-It is a simple yet powerful continuation-based web server, but you can use it without continuations if you like!  
-Check the [documentation](https://docs.racket-lang.org/web-server/) for a quick start guide.  
-There is also a [good tutorial](https://docs.racket-lang.org/continue/) on making a blog written by [Danny Yoo](https://github.com/dyoo) and [Jay McCarthy](https://github.com/jeapostrophe).  
+<p align="left">
+  <a href="https://github.com/racket/web-server">
```
Comment (samth) [inline README.md]: Should this point to the CI results?

**Comment by samth** [comment]
Should the CI use a snapshot? Testing on the most recent release is reasonable but could go wrong in a couple ways.

File: README.md (hunk 1 above)
Comment (Bogdanp) [reply README.md]: Yup. I'll fix it in my next PR.

**Comment by Bogdanp** [comment]
> Should the CI use a snapshot? Testing on the most recent release is reasonable but could go wrong in a couple ways.

I'll make a follow-up PR this week to set up a test matrix and have all the tests run against regular and CS snapshots as well as regular versions and CS versions of 7.3 and 7.4. Normally, this would be pretty easy, but [setup-racket](https://github.com/Bogdanp/setup-racket) doesn't yet support snapshot builds of Racket so I'll have to add that first.

**Pull request merged**